### DISCUSSION TITLE
For connect requests (not gRPC), is it possible to output snake case json?

### DISCUSSION THREAD

**Discussion by JPFrancoia** [connectrpc/connect-go#794] [Q&A]
Hi 👋 

I've started using Connect in an enterpise setting, and there is one thing that could potentially be a problem for me.
At the moment, I have some gRPC clients and some non-gRPC clients hitting my service. The non-gRPC clients send classic http requests, and expect a classic json response. As of now, the json uses camelCase. I would potentially like to use snake case for these classic http requests, mostly for "conformance and homogeneity" with the rest of the company.
Is there a way to achieve that? Maybe an interceptor?

Best

**Comment by jhump** [comment]
@JPFrancoia, this is currently possible via a custom codec. There is a high-quality implementation that provides codecs based on the standard `proto` and `protojson` packages here: https://github.com/akshayjshah/connectproto. You can then use `protojson.MarshalOptions` to use the field's original names, instead of the "JSON-ified" camel-case name.

We are still figuring out just how much surface area we're willing to add to this core package. We highly value that it is small and simple and that most of these things can be easily done in third party packages. But I also realize that many users want a few more features so that this package feels a little more "batteries included". So incorporating things like the above package directly into the connect-go package is still being considered.
